pkg/inclusion/get_commit.go

```diff
@@ -4,9 +4,26 @@ import (
 	"errors"
 
 	"github.com/celestiaorg/celestia-app/pkg/da"
+	"github.com/celestiaorg/celestia-app/pkg/shares"
 	"github.com/tendermint/tendermint/crypto/merkle"
 )
 
+func GetMultiCommit(cacher *EDSSubTreeRootCacher, dah da.DataAvailabilityHeader, startIndexes []uint32, lengths []uint64) ([]byte, error) {
+	shareCounts := make([]int, len(lengths))
+	for i := range lengths {
+		shareCounts[i] = shares.SparseSharesNeeded(uint32(lengths[i]))
+	}
+	commitments := make([][]byte, len(startIndexes))
+	for i := range startIndexes {
+		commitment, err := GetCommit(cacher, dah, int(startIndexes[i]), shareCounts[i])
+		if err != nil {
+			return nil, err
+		}
+		commitments[i] = commitment
+	}
+	return merkle.HashFromByteSlices(commitments), nil
```

Comment:
What's the advantage of merkelizing all the commitments over plain hashing

Reply:
good question, so if we have many blobs being paid for by a PFB, then there might be some scenario where you want to prove that someone paid for a specific blob without downloading the entirely blob. other than that, I'm not really sure. Using a hash would probably work fine

Reply:
> want to prove that someone paid for a specific blob without downloading the entirely blob

Won't you still need the blob or at least the commitment for that blob? 

(I'm generally a bit hesitant about merkelizing something without a strong use case. For example, tendermint merkelizes the header which I always felt was pointless. The headers is like 400 bytes. Just download the entire thing)

Reply:
another good point.

it would be good to get @nashqueue 's point of view on this as well for inclusion fraud proofs. We can discuss synchronously if necessary, as we should probably make a decision soon. I'm definitely fine with changing this to simply include all of the share commitments in the PFB instead of using a secondary commitment over all of them.

Reply:
I'm fairly certain that we would have to end up downloading all of the share commitments for a single PFB if we're attempting to prove inclusion for a single one in both of the other options, those being taking a hash over all of the commitments, or including all commitments in the PFB.

Reply:
If there is ever a scenario where a single PFB pays for 1000 blobs, then I could see the secondary commitment being useful, until then, probably not.

Reply:
I think you make a tradeoff between fraud-proof size and DA bytes. In a way, you can see it if you have multiple commitments, you only need to show that one of them is broken. If you have multiple, you need all blobs to compute the merkle hash and compare that it is wrong. If there is no limit to PFBs per Tx then the worst case fraud proof size for a single commitment fraud-proof would be O(n), with n being the number of shares in a square. This is the case if you have 1 share-sized blob per Tx. So, therefore, I would say saving all commitments is the better option here.

Reply:
sounds good, that's what we'll do then

Reply:
left an issue to handle after this is merged https://github.com/celestiaorg/celestia-app/issues/1231